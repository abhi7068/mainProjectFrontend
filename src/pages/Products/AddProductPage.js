import React, { Component } from 'react';
import AddProduct from '../../organisms/Products/AddProduct';
export default class AddProductPage extends Component {
  render() {
    return (
      <div>
        <AddProduct />
      </div>
    );
  }
}
