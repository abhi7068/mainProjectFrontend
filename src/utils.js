import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
