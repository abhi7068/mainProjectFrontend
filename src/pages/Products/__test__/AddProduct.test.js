import React from 'react';

import AddProductPage from '../AddProductPage';
import { render } from '../../../utils';

it('it renders', () => {
  render(<AddProductPage />);
});
it('should take a snapshot', () => {
  const { asFragment } = render(<AddProductPage />);
  expect(asFragment(render(<AddProductPage />))).toMatchSnapshot();
});
