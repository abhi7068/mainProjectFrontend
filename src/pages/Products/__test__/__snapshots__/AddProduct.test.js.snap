// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="MuiPaper-root MuiCard-root index-root-211 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
            >
              ADD PRODUCT
            </span>
          </div>
        </div>
        <form>
          <div>
            <div
              class="MuiFormControl-root MuiTextField-root"
              style="width: 225px;"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="outlined-basic"
                id="outlined-basic-label"
              >
                Product Name
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="outlined-basic"
                  name="product_name"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-338 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-340"
                  >
                    <span>
                      Product Name
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <br />
            <br />
            <div
              class="MuiFormControl-root MuiTextField-root"
              style="width: 225px;"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="outlined-basic"
                id="outlined-basic-label"
              >
                 Description
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="outlined-basic"
                  name="description"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-338 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-340"
                  >
                    <span>
                       Description
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <br />
            <br />
            <div
              class="MuiFormControl-root MuiTextField-root"
              style="width: 225px;"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink="false"
                for="outlined-basic"
                id="outlined-basic-label"
              >
                Price
                <span
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="outlined-basic"
                  name="price"
                  required=""
                  type="Number"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-338 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-340"
                  >
                    <span>
                      Price *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <br />
            <br />
            <div
              class="MuiFormControl-root MuiTextField-root"
              style="width: 225px;"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink="false"
                for="outlined-basic"
                id="outlined-basic-label"
              >
                Discount
                <span
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="outlined-basic"
                  max="100"
                  min="0"
                  name="discount"
                  required=""
                  type="Number"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-338 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-340"
                  >
                    <span>
                      Discount *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiFormControl-root index-formControl-214"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              Category
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby=" mui-component-select-category"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="mui-component-select-category"
                role="button"
                tabindex="0"
              >
                <span>
                  ​
                </span>
              </div>
              <input
                name="category"
                type="hidden"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <br />
          <br />
          <div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              data-testid="button"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Submit
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
